### PR TITLE
Validate knotx distribution structure

### PR DESCRIPTION
**Pull request by Skejven**
Structure of Knot.x Stack is validated now.
Additionally ZIP archive of stack can now have any name for the root level folder (before only `knotx` was valid).

### CODE REVIEW

File: libraries/_resoruce_helpers.rb [hunk 1]
```diff
@@ -107,6 +107,42 @@ def cp_r(src, dst)
       ::FileUtils.chown_R(node['knotx']['user'], node['knotx']['group'], dst)
     end
 
+    def get_top_level_distro_directory(tmp_dir_path)
```
Comment (jwadolowski) [inline libraries/_resoruce_helpers.rb]: I'd rename that to `get_top_level_distro_directories` as there may be more than 1

File: libraries/_resoruce_helpers.rb [hunk 2]
```diff
@@ -107,6 +107,42 @@ def cp_r(src, dst)
...
+      if top_folder.size == 1
+        tmp_lib_path = ::File.join(tmp_dir_path, top_folder, 'lib')
+        tmp_lib_exists = ::File.exist? tmp_lib_path
+
+        tmp_conf_path = ::File.join(tmp_dir_path, top_folder, 'conf')
+        tmp_conf_exists = ::File.exist? tmp_conf_path
+
+        if !(tmp_lib_exists && tmp_conf_exists)
+          Chef::Application.fatal!(
+            "Expected `lib` and `conf` folders inside #{top_folder} not found!"
```
Comment (jwadolowski) [inline libraries/_resoruce_helpers.rb]: Minor thing, but backticks won't be evaluated in this context

Comment (Skejven) [reply libraries/_resoruce_helpers.rb]: I don't want to evaluate them, this is just my habit form markdown. I will remove backticks from here.

File: libraries/_resoruce_helpers.rb (hunk 1 above)
Comment (Skejven) [reply libraries/_resoruce_helpers.rb]: I will change `select` to `find`.

**Pull request merged**